Python/ceval_gil.c

```diff
@@ -547,9 +547,11 @@ _PyEval_FiniGIL(PyInterpreterState *interp)
         return;
     }
     else if (!interp->ceval.own_gil) {
+#ifdef Py_DEBUG
         PyInterpreterState *main_interp = _PyInterpreterState_Main();
         assert(interp != main_interp);
         assert(interp->ceval.gil == main_interp->ceval.gil);
+#endif
```

Comment:
`_PyInterpreterState_Main` is declared as a trivial inlinable function: https://github.com/python/cpython/blob/f5088006ca8e9654fbc3de119462f0ab764e408b/Include/internal/pycore_pystate.h#L25-L28

Would it be better to get rid of the variable altogether?

```suggestion
        assert(interp != _PyInterpreterState_Main());
        assert(interp->ceval.gil == _PyInterpreterState_Main()->ceval.gil);
```

Reply:
I think that our standard `Py_DEBUG` hack is better :)